packages/core/src/index.ts

```diff
@@ -32,5 +35,14 @@ export function analyzeProject(projectDirectory: string = process.cwd()): Projec
   let languageServer = new GlintLanguageServer(glintConfig, documents, transformManager);
   let shutdown = (): void => languageServer.dispose();
 
-  return { glintConfig, transformManager, languageServer, shutdown };
+  return {
+    glintConfig,
+    transformManager,
+    languageServer,
+    documents,
```

Comment:
The `documents` export here gives me the heebie-jeebies. It's very much an internal detail of how the system works, and not (I think?) a thing we want to expose as something to generally support. Maybe a better approach here would be to define what it is a consumer *needs* that the `DocumentCache` provides a way of getting, and provide a function here which can answer that question (but in a way we can evolve over time)? 🤔

Reply:
That's a fair concern, for sure.

IME we really only need `getDocumentVersion`, `getDocumentContents`, and `updateDocument`, which could easily just be three methods on a POJO that gets returned from `analyzeProject`. Would that work for you?

Reply:
That definitely seems *more* reasonable! Can you expand a bit on what each of those is used for? 🙏🏼

Reply:
Of course!

All three of these methods are used to satisfy parts of the TypeScript `LanguageServerHost` interface (e.g. `getScriptSnapshot`).

`getDocumentContents` and `updateDocument` are both used to (as you might guess) interact with documents that Glint has stored. This comes in handy since we are essentially using Glint to analyze and (sometimes) fix issues found throughout a codebase while converting it to TypeScript, so the ability to, for example, pull a document's contents out of Glint,  apply a codefix to it, and then write it back to the document cache comes in very handy.

We don't actually use `getDocumentVersion` directly, but it may get called by other parts of the TS  Compiler API as part of the language host.

Reply:
Makes sense to me; I'll let @dfreeman confirm, though!

Reply:
I'm a little confused about the usage pattern here—from the fact that you're needing to construct a `LanguageServiceHost`, it sounds like you're spinning up a second `LanguageService` in _addition_ to the one Glint instantiates when you run `analyzeProject`?